### PR TITLE
Bugfix FXIOS-10403 - Application crashes when closing a tab in tabs tray on certain devices

### PR DESCRIPTION
**Pull request by PARAIPAN9**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-10403)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/22789)

## :bulb: Description

- Remove `TabTrayAnimationQueue` file that was used for performing animations.
- No need of manual animations when using `UICollectionViewDiffableDataSource`.
## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/Client/Frontend/Browser/Tabs/Views/TabDisplayView.swift [hunk 1]
```diff
@@ -78,14 +77,12 @@ class TabDisplayView: UIView,
 
     public init(panelType: TabTrayPanelType,
                 state: TabsPanelState,
-                windowUUID: WindowUUID,
-                animationQueue: TabTrayAnimationQueue = TabTrayAnimationQueueImplementation()) {
```
Comment (mattreaganmozilla) [inline firefox-ios/Client/Frontend/Browser/Tabs/Views/TabDisplayView.swift:82]: At a glance it looks like this is the only place that `TabTrayAnimationQueueImplementation` was used? NAB for this PR but we could probably also remove the related `TabTrayAnimationQueue.swift` file.

**Review by mattreaganmozilla — APPROVED** [review]
@OrlaM and @nbhasin2 are much more familiar with this area than me but changes lgtm.

**Review by dataports — APPROVED** [review]
Looks good to me! I meant to remove these but totally forgot, thank you @PARAIPAN9 and thanks for the tag @OrlaM

**Comment by dataports** [comment]
> I don't remember the context of what issue it was that drove us to add the animation queue. @dataports can you take a look and verify this doesn't regress any of the issues you have been fixing in this area?

Verified, the animation queue was trying to clean up the animations on the tab tray without diffable datasource but it's fixed now that it's been introduced, so it's no longer necessary.

**Comment by nbhasin2** [comment]
cc @dataports any issues around backporting to v133?

**Comment by nbhasin2** [comment]
@mergify backport release/v133

**Comment by dataports** [comment]
> cc @dataports any issues around backporting to v133?

The diffable datasource changes are already in v133 so should be fine

**Pull request merged**